internal/service/users/user.go

```diff
@@ -54,6 +55,28 @@ func (s *Service) AllUsers() ([]openapi.UserData, error) {
 	return usersData, nil
 }
 
+// Get or create a user for a unique username
+// If they do not exist then they will be created with the base role
+func (s *Service) PersistedUser(username types.Username) (types.User, error) {
+	user := types.User{}
+	result := s.db.Where("username = ?", username).
+		Attrs(types.User{
+			Username: username,
+			Model:    types.Model{CreatedAt: time.Now()},
+		}).
+		FirstOrCreate(&user)
+	if result.Error != nil {
+		return user, types.NewErrServerError(result.Error)
+	}
+	userWasCreated := result.RowsAffected > 0
+	if userWasCreated {
+		if _, err := rbac.AddRole(user, rbac.Base); err != nil {
+			return user, fmt.Errorf("failed assign user base role: %v", err)
+		}
+	}
+	return user, nil
```

Comment:
can we also return userWasCreated so it can be used in other conditionals after 🙏